service/history/hsm/tree.go

```diff
@@ -61,6 +61,12 @@ type StateMachineDefinition interface {
 	Serialize(any) ([]byte, error)
 	// Deserialize a state machine from bytes.
 	Deserialize([]byte) (any, error)
+	// CompareState compares two state objects. It should return 0 if the states are equal, a positive number if the
```

Comment:
I left this as 1 and -1 in my PR to be consistent with function like strings.Compare(). 
Now that I read you impl, I understand why you want it to be a positive/negative number :)